packages/dbs-ai/server/lib/Smarti.js

```diff
@@ -0,0 +1,192 @@
+/* globals _dbs, SystemLogger, RocketChat */
+
+class SmartiAdapter {
+	constructor(adapterProps) {
+
+		this.properties = adapterProps;
+		this.properties.url = this.properties.url.toLowerCase();
+
+		this.options = {};
+		this.options.headers = {};
+		this.options.headers['content-Type'] = 'application/json; charset=utf-8';
+		if (this.properties.token) {
+			this.options.headers['authorization'] = `basic ${ this.properties.token }`;
+		}
+		if (this.properties.url.substring(0, 4) === 'https') {
+			this.options.cert = `~/.nodeCaCerts/${ this.properties.url.replace('https', '') }`;
+		}
+	}
+
+	onMessage(message) {
+
+		//TODO is this always a new one, what about update
+
+		const requestBody = {
+			webhook_url: this.properties.webhookUrl,
+			message_id: message._id,
+			channel_id: message.rid,
+			user_id: message.u._id,
+			username: message.u.username,
+			text: message.msg,
+			timestamp: message.ts,
+			origin: message.origin
+		};
+
+		try {
+			const options = this.options;
+			this.options.data = requestBody;
+
+			SystemLogger.debug('Smarti - trigger analysis:', JSON.stringify(options, '', 2));
+
+			const response = HTTP.post(`${ this.properties.url }rocket/${RocketChat.settings.get('DBS_AI_Redlink_Domain')}`, options);
+
+			if (response.statusCode === 200) {
+				SystemLogger.debug('Smarti - analysis triggered successfully:', JSON.stringify(response, '', 2));
+			} else {
+				SystemLogger.error('Smarti - analysis triggering failed:', JSON.stringify(response, '', 2));
+			}
+		} catch (e) {
+			SystemLogger.error('Smarti response did not succeed: ', e);
+		}
+	}
+
+	onClose(room) { //async
+		//TODO add options here?
```

Comment:
@tkurz which options do you mean? Header options?
What about transmitting a context?